x/epochs/client/cli/query_test.go

```diff
@@ -0,0 +1,74 @@
+package cli_test
+
+import (
+	gocontext "context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/v12/app/apptesting"
+	"github.com/osmosis-labs/osmosis/v12/x/epochs/types"
+)
+
+type QueryTestSuite struct {
+	apptesting.KeeperTestHelper
+	queryClient types.QueryClient
+}
+
+func (s *QueryTestSuite) SetupSuite() {
+	s.Setup()
+	s.queryClient = types.NewQueryClient(s.QueryHelper)
+
+	// add new epoch
+	epoch := types.EpochInfo{
+		Identifier:              "weekly",
+		StartTime:               time.Time{},
+		Duration:                time.Hour,
+		CurrentEpoch:            0,
+		CurrentEpochStartHeight: 0,
+		CurrentEpochStartTime:   time.Time{},
+		EpochCountingStarted:    false,
+	}
+	s.App.EpochsKeeper.AddEpochInfo(s.Ctx, epoch)
+
+	s.Commit()
+}
+
+func (s *QueryTestSuite) TestQueriesNeverAlterState() {
+	s.SetupSuite()
+	testCases := []struct {
+		name  string
+		query func()
+	}{
+		{
+			"Query current epoch",
+			func() {
+				res, err := s.queryClient.CurrentEpoch(gocontext.Background(), &types.QueryCurrentEpochRequest{Identifier: "weekly"})
```

Comment:
nit: Let's try to avoid these callbacks in table-driven test cases if possible please. While this test case is not complex, for bigger test cases this pattern becomes confusing. I think having a completely separate test fixture is preferable to avoid encouraging this style.

Reply:
I know it's a lot to refactor so I discussed this with Matt, and he agrees with the suggestion to split it up if possible, please

Reply:
So should split each query into a separate test function?

Reply:
To avoid table-driven callbacks, I think there is another way to use queryClient.cc.Invoke and pass query path, request,...

Reply:
I think that sounds reasonable. Do you mind changing one test to see how it compares?

Reply:
Thanks, IMO the epoch change looks good.

Reply:
Thank you, I'll apply it to other modules